### PR TITLE
Quick fix: Address a couple of nits in / around `ViewDataDictionary`

### PR DESCRIPTION
**Pull request by dougbu**
- remove duplicate `null` checks in `ViewDataDictionary` constructors
- correct a couple of code comments
- use a `public` constructor for `ViewDataDictionary` in `ViewExecutor`

### CODE REVIEW

**Comment by dougbu** [comment]
Mostly FYI but @ryanbrandenburg please glance through.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewExecutor.cs [hunk 1]
```diff
@@ -130,11 +130,11 @@ public class ViewExecutor
                 throw new ArgumentNullException(nameof(view));
             }
 
-            var services = actionContext.HttpContext.RequestServices;
             if (viewData == null)
             {
+                var services = actionContext.HttpContext.RequestServices;
                 var metadataProvider = services.GetRequiredService<IModelMetadataProvider>();
-                viewData = new ViewDataDictionary(metadataProvider);
+                viewData = new ViewDataDictionary(metadataProvider, actionContext.ModelState);
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewExecutor.cs]: `ModelState` is a get-only property that can never be `null`. Therefore, no need to call the `internal` constructor.

**Comment by ryanbrandenburg** [comment]
This all seems legit. :shipit:

**Comment by dougbu** [comment]
384fd1f

**Pull request closed without merging**